pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/design/CognitiveComplexityTest.java

```diff
@@ -1,10 +0,0 @@
-/**
- * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
- */
-
-package net.sourceforge.pmd.lang.java.rule.design;
-
-import net.sourceforge.pmd.testframework.PmdRuleTst;
-
-public class CognitiveComplexityTest extends PmdRuleTst {
```

Comment:
This file actually needs to be there. We have separate tests for the metric and the rule, there needs to be both.

Reply:
I've added SonarSource examples as a rule test - should be good as acceptance test. Please have a look.

Reply:
Yes this is fine :)